.github/workflows/linkchecker.yml

```diff
@@ -0,0 +1,22 @@
+name: Broken link check
+
+on:
+  workflow_run:
+    workflows: ["build-and-deploy"]
+    types:
+      - completed
+
+jobs:
+  broken_link_checker_job:
+    runs-on: ubuntu-latest
+    name: Check for broken links
+    steps:
+    - name: Check for broken links
+      id: link-report
+      uses: celinekurpershoek/github-actions-link-checker@{version}
+      with:
+        url: 'https://devonfw.com' 
```

Comment:
Along with `url`, the action has a parameter called `recursiveLinks` to perform link checking on the entire site. I think you should use it and set it to `true`.